src/Glpi/Marketplace/View.php

```diff
@@ -268,6 +268,46 @@ public static function installed(
         self::displayList($plugins, "installed", $only_lis);
     }
 
+    /**
+     * Get the logo of a plugin
+     * Useful for custom plugins which will be used internally only and are not published in the internet
+     *
+     * @param array   $apidata  collected data of the plugin
+     * @param string  $key      plugins directory name
+     *
+     * @return string url to the plugins logo or empty string
+     */
+    public static function getLogo(array $apidata, string $key): string
+    {
+
+        // found a logo from online resource?
+        if (isset($apidata['logo_url']) && strpos(strtolower($apidata['logo_url']), '.png') !== false) {
+            return $apidata['logo_url'];
+        }
+
+        // path to the plugin directory (for checking if a local file exists)
+        $pluginDir = Plugin::getPhpDir($key);
+
+        // try to find a local logo
+        $logo = $pluginDir . $key . '.png';
+
+        // logo named as the key found?
+        if (file_exists($logo) && is_readable($logo)) {
```

Comment:
There should be a check that the file is actually an image I think. Not sure it is a big issue, but should be safer.